internal/backend/sftp/sshcmd_test.go

```diff
@@ -30,6 +30,21 @@ var sshcmdTests = []struct {
 		"ssh",
 		[]string{"host", "-p", "10022", "-l", "user", "-s", "sftp"},
```

Comment:
I think we should remove this test (and support for passing a `host:port` string to `buildSSHCommand()`. That caused us problems in the past, to the point that we removed parsing the port at all, and added hints to the manual how users can work around this: https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html#sftp

Reply:
I don't think this is a step that should be taken. Again, making the repository parsing more robust is my preference.

All of this becomes less important when `restic` has it's own configuration file support. However, even then when you have to run `restic` as different users or on many systems, being able to specify all of the configuration in a single command line is much better. I don't like the idea of needing to ensure a specific configuration across many files or users is properly setup before running `restic` operations.

That documentation should include a working example of `-o sftp.command="foobar"` and possibly also the default sftp command that's built.

The options to be built appear to be these:
- `-o sftp.command="ssh host -p 22 -l user -s sftp"`
- `-o sftp.command="ssh host -p 22 -s sftp"`
- `-o sftp.command="ssh host -l user -s sftp"`
- `-o sftp.command="ssh host -s sftp"`

I know I recently went looking for how to get `restic` to use `krssh` and this information would have been helpful then. I ended up creating a wrapper named `ssh` then changing the `PATH` `restic` was using so that this wrapper was found first.